charmcraft/commands/init.py

```diff
@@ -133,7 +133,7 @@ def run(self, args):
                 fh.write(out)
                 for todo in _todo_rx.findall(out):
                     todos.append((template_name, todo))
-                if template_name in executables:
+                if template_name in executables and os.name == "posix":
```

Comment:
It's simpler and faster to check if `posix` above when building the `executables` list, where it's also a good place to indicate what's going on with a comment, like "we will set these to be executables if posix only, as other platforms doesn't support it".

Also missing a test.

Reply:
The relevant tests update is actually in another PR:
https://github.com/canonical/charmcraft/pull/516/files#diff-134db82b983286583fedb40b7d946fd37d08efba163ea4b54e36ee1628f004f3R133